File: Go/num_of_good_pairs/1512.go
package numofgoodpairs

func numIdenticalPairs(nums []int) int {
	result := 0
	numberOfCountMap := make(map[int]int)
	for _, num := range nums {
		if count, ok := numberOfCountMap[num]; ok {
			result += count
			numberOfCountMap[num] = count + 1
		} else {
			numberOfCountMap[num] = 1
		}
	}
	return result
}

File: TypeScript/num_of_good_pairs/1512.ts
/**
 * Runtime: 62ms (Beats 51.8%)
 * Memory: 43MB (Beats 41.44%)
 */

function numIdenticalPairs(nums: number[]): number {
  let result = 0;
  const hashMap = new Map<number, number>();
  for (const num of nums) {
    if (!hashMap.has(num)) {
      hashMap.set(num, 1);
    } else {
      result += hashMap.get(num) ?? 0;
      hashMap.set(num, (hashMap.get(num) ?? 0) + 1);
    }
  }
  return result;
}

export default numIdenticalPairs;
